src/utils/validationUtil.ts

```diff
@@ -47,119 +66,113 @@ export class ValidationUtil {
-            if (
-              testType.testResult === enums.TEST_RESULT.PASS &&
-              payload.testStatus === enums.TEST_STATUS.SUBMITTED
-            ) {
-              if (!testType.testExpiryDate) {
-                missingFields.push(enums.ERRORS.NoLECExpiryDate);
-              }
-              if (!testType.modType) {
-                missingFields.push(enums.ERRORS.NoModificationType);
-              }
-              if (!testType.emissionStandard) {
-                missingFields.push(enums.ERRORS.NoEmissionStandard);
-              }
-              if (!testType.fuelType) {
-                missingFields.push(enums.ERRORS.NoFuelType);
-              }
-              if (!testType.smokeTestKLimitApplied) {
-                missingFields.push(enums.ERRORS.NoSmokeTestKLimitApplied);
-              }
-            }
-          }
+    const { testTypes, testStatus } = payload;
+    testTypes
+      .filter((testType) => ValidationUtil.isTestTypeLec(testType))
+      .filter(
+        (testType) =>
+          testType.testResult === enums.TEST_RESULT.PASS &&
+          testStatus === enums.TEST_STATUS.SUBMITTED
+      )
+      .forEach((testType) => {
+        const {
+          testExpiryDate,
+          modType,
+          emissionStandard,
+          fuelType,
+          smokeTestKLimitApplied,
+        } = testType;
+        if (!testExpiryDate) {
+          missingFields.push(enums.ERRORS.NoLECExpiryDate);
         }
-      );
-    }
+        if (!modType) {
+          missingFields.push(enums.ERRORS.NoModificationType);
+        }
+        if (!emissionStandard) {
+          missingFields.push(enums.ERRORS.NoEmissionStandard);
+        }
+        if (!fuelType) {
+          missingFields.push(enums.ERRORS.NoFuelType);
+        }
+        if (!smokeTestKLimitApplied) {
+          missingFields.push(enums.ERRORS.NoSmokeTestKLimitApplied);
+        }
+      });
     return missingFields;
   }
 
   public static validateMandatoryTestResultFields(
     payload: models.ITestResultPayload
   ): string[] {
     const missingMandatoryFields: string[] = [];
+    const {
+      testTypes,
+      testStatus,
+      countryOfRegistration,
+      euVehicleCategory,
+      vehicleType,
+      odometerReading,
+      odometerReadingUnits,
+    } = payload;
     if (
-      payload.testTypes.some(
+      testTypes.every(
         (testType: models.TestType) =>
-          testType.testResult !== enums.TEST_RESULT.ABANDONED
-      ) &&
-      payload.testStatus === enums.TEST_STATUS.SUBMITTED
+          testType.testResult === enums.TEST_RESULT.ABANDONED
+      ) ||
+      testStatus !== enums.TEST_STATUS.SUBMITTED
     ) {
-      if (!payload.countryOfRegistration) {
-        missingMandatoryFields.push(
-          enums.ERRORS.CountryOfRegistrationMandatory
-        );
-      }
-      if (!payload.euVehicleCategory) {
-        missingMandatoryFields.push(enums.ERRORS.EuVehicleCategoryMandatory);
-      }
+      return missingMandatoryFields;
```

Comment:
I cannot see where you add the missing country of registration error in case there are no abandoned or cancelled tests